server/pbench/bin/index-pbench

```diff
@@ -804,126 +1133,135 @@ def mk_results_action(hostname, tbname, dirname, dirprefix, tb, mdconf, idxname,
-    yield mk_results_action(hostname, tbname, dirname, dirprefix, tb, mdconf, idxname, md5sum, tstamp, options)
+    yield mk_results_action(ptb, options, INDEX_PREFIX)
     if debug_time_operations: ts("mk_toc_actions")
-    for action in mk_toc_actions(hostname, tbname, dirname, dirprefix, tb, mdconf, idxname, md5sum, tstamp, options):
+    for action in mk_toc_actions(ptb, options, INDEX_PREFIX):
         yield action
     if debug_time_operations: ts("mk_tool_data_actions")
-    for action in mk_tool_data_actions(hostname, tbname, dirname, dirprefix, tb, mdconf, idxname, md5sum, tstamp, options):
+    for action in mk_tool_data_actions(ptb, options, INDEX_PREFIX):
         yield action
     if debug_time_operations: ts("", newline=True)
     return
 
-###########################################################################
-# We get the start date out of the metadata log.
 
-class BadDate(Exception):
-    pass
+class PbenchTarBall(object):
+    def __init__(self, tbarg):
+        global _tmpdir
 
+        self.tbname = tbarg
+        self.hostname = os.path.basename(os.path.dirname(self.tbname))
+        self.tb = tarfile.open(self.tbname)
 
-def run_start_date(mdconf, tbname):
-    # import pdb; pdb.set_trace()
-    try:
-        # N.B. the start_run timestamp is in UTC.
-        tstamp = mdconf.get('run', 'start_run').split('_')
-        date = [ int(x) for x in tstamp[0].split("-") ]
-    except Exception:
-        raise BadDate(tbname)
-    # for future reference
-    # time = [ int(x) for x in tstamp[1].split(":") ]
-
-    return date
-
-###########################################################################
-# main: create index from template if necessary, prepare the action and ship
-# it to ES for indexing.
-
-# status codes: these are used by pbench-index to segregate tarballs into
-#               classes: should we retry (perhaps after fixing bugs in the
-#               indexing) or should we just forget about them?
-#       0 - normal exit, no errors.
-#       1 - operational error (bad mapping file, no config file).
-#       2 - tarball does not contain a metadata.log file, do not try to index again.
-#       3 - tarball name does not encode the date, do not try to index again.
-#       4 - generic error, needs to be investigated and can be retried after
-#           any indexing bugs are fixed.
-
-class ConfigFileNotSpecified(Exception):
-    pass
-
-
-class ConfigFileError(Exception):
-    pass
+        # This is the top-level name of the run - it should be the common
+        # first component of every member of the tarball.
+        dirname = os.path.basename(self.tbname)
+        # FIXME: should we handle result tarballs that are uncompressed, or
+        # compressed with a different compression mechanism from xz?
+        self.dirname = dirname[:dirname.rfind('.tar.xz')]
+        # ... but let's make sure.
+        # FIXME: how are we making sure it is the first component of every
```

Comment:
@ndokos, can you review this comment and see how this is taken care of properly?  I thought I saw where this is handled, but don't remember any longer.